src/StackExchange.Redis/AzureMaintenanceEvent.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using static StackExchange.Redis.ConnectionMultiplexer;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// Azure node maintenance event. For more information, please see: https://github.com/Azure/AzureCacheForRedis/blob/main/AzureRedisEvents.md
+    /// </summary>
+    public class AzureMaintenanceEvent
+    {
+        private const string PubSubChannelName = "AzureRedisEvents";
+
+        internal AzureMaintenanceEvent(string azureEvent)
+        {
+            if (azureEvent == null)
+            {
+                return;
+            }
+
+            // The message consists of key-value pairs delimted by pipes. For example, a message might look like:
+            // NotificationType|NodeMaintenanceStarting|StartTimeUtc|2021-09-23T12:34:19|IsReplica|False|IpAddress|13.67.42.199|SSLPort|15001|NonSSLPort|13001
+            var message = new ReadOnlySpan<char>(azureEvent.ToCharArray());
+            try
+            {
+                while (message.Length > 0)
+                {
+                    if (message[0] == '|')
+                    {
+                        message = message.Slice(1);
+                        continue;
+                    }
+
+                    // Grab the next pair
+                    var key = message.Slice(0, message.IndexOf('|'));
+                    message = message.Slice(key.Length + 1);
+
+                    var valueEnd = message.IndexOf('|');
+                    var value = valueEnd > -1 ? message.Slice(0, valueEnd) : message;
+                    message = message.Slice(value.Length);
+
+                    if (key.Length > 0 && value.Length > 0)
+                    {
+                        switch (key)
+                        {
+                            case var _ when key.SequenceEqual(nameof(NotificationType).ToCharArray()):
```

Comment:
these `.ToCharArray()` allocations are unnecessary; on modern frameworks (from netcoreapp3.0 onwards, I believe) there is an implicit conversion from `string` to `ReadOnlySpan<char>`, but since we have down-level TFMs in scope, you might need to use `nameof(NotificationType).AsSpan()`

(applies to all similar in this `switch`)